### ISSUE TITLE
Layout/EmptyLinesAroundAccessModifier and Lint/UselessAccessModifier should only apply at class scope

### ISSUE THREAD

**Issue by liamwhite** [rubocop-hq/rubocop#7212]
It is possible to trigger the style cop Layout/EmptyLinesAroundAccessModifier in various ways by using `public` and `private` as instance methods on a class which are not directly defined in that class.

In addition, it is possible to trigger the lint Lint/UselessAccessModifier by using the `public` and `private` access modifiers e.g. in conditionals.

Below is a (somewhat pathological) testcase that showcases these problems.

```rb
# frozen_string_literal: true

# Fails rubocop when it should not
class RubocopTest
  def self.block(&block)
    @block = block if block_given?

    @block.call unless block_given?
  end

  def initialize
    self.class.block
  end

  block do
    if public && ARGV[0] == '1'
      { public: public }
    else
      {}
    end
  end

  instance_eval do
    def public
      puts 'Public!'
    end
  end
end

RubocopTest.new

```

--------

## Expected behavior

No style cops or lint errors should occur. This program outputs `Public!` when it runs.

## Actual behavior

With `0.73.0 (using Parser 2.6.3.0, running on ruby 2.6.3 x86_64-linux)`:

```
Inspecting 1 file
W

Offenses:

rubocop_test.rb:16:8: C: Layout/EmptyLinesAroundAccessModifier: Keep a blank line after public.
    if public && ARGV[0] == '1'
       ^^^^^^
rubocop_test.rb:16:8: W: Lint/UselessAccessModifier: Useless public access modifier.
    if public && ARGV[0] == '1'
       ^^^^^^
rubocop_test.rb:17:17: C: Layout/EmptyLinesAroundAccessModifier: Keep a blank line before and after public.
      { public: public }
                ^^^^^^
rubocop_test.rb:17:17: W: Lint/UselessAccessModifier: Useless public access modifier.
      { public: public }
                ^^^^^^

1 file inspected, 4 offenses detected
```

**Comment by mdole** [comment]
+1, also experiencing this issue with the following code:
```
def private?
  private || private.nil?
end
```

**Comment by rafaelfranca** [comment]
Same:

```ruby
client.write? ? public.write : public.read
```

**Comment by buehmann** [comment]
@koic I used `git bisect` to see when this was introduced: ffeccd639b9f40d755f0523e6e59d10edabf3b83 is the first bad commit
